### ISSUE TITLE
Clock Item Format

### ISSUE THREAD

**Issue by chalamministries** [ChurchApps/FreeShow#1040]
**Describe the feature**
Ability to set the format for the time on the Clock Item you add to a slide. Maybe dropdown of layouts.
Ability to also show the date as well as time

**Comment by chalamministries** [comment]
I'll try to work on my fork to see if I can implement this if you think its worth it

**Comment by vassbo** [comment]
What formats did you have in mind?

For date and also digital time you can use a normal textbox with (right click) "Dynamic values" (It's missing 12h time at the moment).

But go ahead if you think it should be implemented into the clock item! :)

**Comment by chalamministries** [comment]
Well we us it for a TV timer in our studio and we do Thursday November 20, 2024 11:00:00AM

Of course it doesnt have to be full like that but Definitely the month and day. I'll take a look at it. Basically formats listed like whats on the momentjs page:

https://devhints.io/moment

https://devhints.io/moment	https://devhints.io/moment	[Moment.js cheatsheet
The one-page guide to Moment.js: usage, examples, links, snippets, and more.](https://devhints.io/moment)	https://devhints.io/moment

[Steven Gauerke](https://spikenow.com/r/a/?ref=spike-organic-signature&_ts=2vabha)	[2vabha]Web Developer

On November 21, 2024 at 7:00 GMT, Kristoffer Vassbø via Info ***@***.***> wrote:

What formats did you have in mind?

For date and also digital time you can use a normal textbox with (right click) "Dynamic values" (It's missing 12h time at the moment).

But go ahead if you think it should be implemented into the clock item! :)

**Comment by mwb9aa** [comment]
I have a different clock enhancement. I would like the option to show only minutes and seconds. For example, with a timer for service start, I would like the timer to show 90:00 instead of 01:30:00. Maybe this can be a checkbox to choose the HH:MM:SS or MM:SS format.

I would like this because the text width stays more consistent if I can run MM:SS, instead of HH:MM:SS.

@vassbo Would you like me to make a separate enhancement request?

**Comment by chalamministries** [comment]
I'm currently working on the feature updates for clock and timers. I'll have to think about this but currently I'm not sure if that's doable as MM is only minutes up to 60 as programmed with JavaScript datetime.

**Comment by vassbo** [comment]
@mwb9aa That would be doable in the "Timer" item, we can keep this here for now.
@chalamministries If you want to do that you can do it, but if not I'll do it once you have finished updating the clock/timers.

**Comment by chalamministries** [comment]
Kristoffer Vassbø via Info
@mwb9aa That would be doable in the "Timer" item, we can keep this here for now. @chalamministries If you want to do that you can do it, but if not I'll do it once you have finished updating the clock/timers.
Yes ill take a stab at it because for clock and timer, I'm wanting to add a "Format" option. Would you prefer a dropdown of options? Or let them enter the HH:MM:SS format like you do on Date Format in settings?

[Steven Gauerke](https://spikenow.com/r/a/?ref=spike-organic-signature&_ts=2vhtq2)	[2vhtq2]Web Developer

On November 25, 2024 at 7:04 GMT, Kristoffer Vassbø via Info ***@***.***> wrote:

***@***.***(https://github.com/mwb9aa) That would be doable in the "Timer" item, we can keep this here for now.
***@***.***(https://github.com/chalamministries) If you want to do that you can do it, but if not I'll do it once you have finished updating the clock/timers.

**Comment by vassbo** [comment]
Great. Possibly a dropdown for the most common values, and the last option in the dropdown would be "Custom" where you can enter a value like HH:MM:SS in a textbox!

**Comment by chalamministries** [comment]
@mwb9aa I know you are asking about a different feature, but would you like to share your input on the clock date/time format? Do you think simple dropdowns with some predefined formats? Or do you think users will know how to use the HH:mm:ss formating?

**Comment by mwb9aa** [comment]
@chalamministries I don't think it matters much. A dropdown is most accessible, but I think the HH:mm:ss style writing is very common practice, and would just need a note in the documentation. I like what @vassbro put [here](https://github.com/ChurchApps/FreeShow/issues/1040#issuecomment-2499806444). I think that gets the best of both worlds. Having said that JavaScript is not good with a variety of date formats. I'm not sure how Electron or Svelte handle them (if at all). [PHP](https://www.php.net/manual/en/class.datetimeinterface.php#datetimeinterface.constants.types) handles a variety of formats.

My preference is that all numerical values have a leading zero, but in my area, I think the general populace prefers no leading zero (02 instead of 2).

Dropdown Date/Time items:

- Month Day Year Time /12 (numerical)
- Month Day Year Time /24 (numerical)
- Day Month Year Time /12 (numerical)
- Day Month Year Time /24 (numerical)
- Month Day Year Time /12 (textual)
- Month Day Year Time /24 (textual)
- Day Month Year Time /12 (textual)
- Day Month Year Time /24 (textual)
- Custom Format:

Dropdown Timer items:

- hour:minutes:seconds /12
- hour:minutes:seconds /24
- minutes:seconds
- seconds
- Custom Format:

As big as the dropdown list is, it might be easier/faster/cleaner just to make a note in the documentation as to the formats and have a single default format (like it is now), but allow a user to edit the format with an option in the timer/date create/edit box.

- [ ] Custom Format   `<input type="text" ... placeholder="Type custom format. Please see the documentation for accepted formats.">`

**Comment by chalamministries** [comment]
Just letting everyone know, I'm still working on this.
